core/sqf/hbase_utilities/src/main/java/org/trafodion/utility/backuprestore/TrafExportSnapshot.java

```diff
@@ -0,0 +1,1076 @@
+      // Check if the snapshot already in-progress
+      if (outputFs.exists(snapshotTmpDir)) {
+        if (overwrite) {
+          if (!outputFs.delete(snapshotTmpDir, true)) {
+            System.err.println("Unable to remove existing snapshot tmp directory: "+snapshotTmpDir);
+            return 1;
+          }
+        } else {
+          System.err.println("A snapshot with the same name '"+ targetName +"' may be in-progress");
+          System.err.println("Please check "+snapshotTmpDir+". If the snapshot has completed, ");
+          System.err.println("consider removing "+snapshotTmpDir+" by using the -overwrite option");
+          return 1;
+        }
+      }
+    }
+
+    // Step 1 - Copy fs1:/.snapshot/<snapshot> to  fs2:/.snapshot/.tmp/<snapshot>
+    // The snapshot references must be copied before the hfiles otherwise the cleaner
+    // will remove them because they are unreferenced.
+    try {
+      LOG.info("Copy Snapshot Manifest");
+      FileUtil.copy(inputFs, snapshotDir, outputFs, initialOutputSnapshotDir, false, false, conf);
+    } catch (IOException e) {
+      throw new ExportSnapshotException("Failed to copy the snapshot directory: from=" +
+        snapshotDir + " to=" + initialOutputSnapshotDir, e);
+    }
+
+    // Write a new .snapshotinfo if the target name is different from the source name
+    if (!targetName.equals(snapshotName)) {
+      SnapshotDescription snapshotDesc =
+        SnapshotDescriptionUtils.readSnapshotInfo(inputFs, snapshotDir)
+          .toBuilder()
+          .setName(targetName)
+          .build();
+      SnapshotDescriptionUtils.writeSnapshotInfo(snapshotDesc, snapshotTmpDir, outputFs);
+    }
+
+    List<Pair<SnapshotFileInfo, Long>> snapshotFiles = getSnapshotFiles(conf, inputFs, snapshotDir);
+    Long snapshotTotalDataSize = 0L;
+    boolean hasWALfiles = false;
+    SnapshotFileInfo inputInfo;
+    for (Pair<SnapshotFileInfo, Long> fileInfo: snapshotFiles) {
+    	snapshotTotalDataSize += fileInfo.getSecond();
+    	if (fileInfo.getFirst().getType() == SnapshotFileInfo.Type.WAL)
+    		hasWALfiles = true;
+    }
+    
+    
+
+    // Step 2 - Start MR Job to copy files
+    // The snapshot references must be copied before the files otherwise the files gets removed
+    // by the HFileArchiver, since they have no references.
+    try {
+    	if ((snapshotTotalDataSize/(1024*1024) > mrLowLimitMB) || hasWALfiles) {
+    	    runCopyJob(inputRoot, outputRoot, snapshotName, snapshotDir, verifyChecksum,
+    		       filesUser, filesGroup, filesMode, mappers, bandwidthMB);
+    	}
+    	else {
+    	     	LOG.info("Copy Snapshot Data Files");
+    	     	for (Pair<SnapshotFileInfo, Long> fileInfo: snapshotFiles) {
+    	     		inputInfo = fileInfo.getFirst();
+    	     		Path outputPath = getCopyOutputPath(conf, inputInfo);
+    	     		Path inputPath = getCopyInputPath(conf, inputInfo, inputFs);
+    	     		FileUtil.copy(inputFs, inputPath, outputFs, outputPath, false, false, conf);
```

Comment:
Should we consider setting the filesuser, filesGroup and filesMode on the target cluster similar to runCopyJob. This may be needed if we need to Export  a single table using this utility in future.